### PR TITLE
minitel_2_rpic.cpp : Modem and serial port support

### PR DESCRIPTION
**Pull request by jfdelnero**
minitel_2_rpic.cpp : Modem and serial port support.

Modem and external port can be exported to the "modport0" and "serport0" interfaces.
Example : Command line options to use to create a TCP socket linked to the modem port : "-modport0 null_modem -bitb socket.127.0.0.1:20000". Once mame started you can then send vdt files with netcat to this socket.

### CODE REVIEW

File: src/mame/drivers/minitel_2_rpic.cpp [hunk 1]
```diff
@@ -60,28 +72,32 @@
...
+#define PORT_1_SCL_BIT         6
+#define PORT_1_SDA_BIT         7
+
+#define PORT_3_SER_RXD_BIT     0
+#define PORT_3_SER_TXD_BIT     1
+#define PORT_3_SER_ZCO_BIT     2
+#define PORT_3_MDM_RXD_BIT     3
+#define PORT_3_SER_RDY_BIT     5
+
+#define MASK(bit) (0x01 << bit)
```
Comment (cuavas) [inline src/mame/drivers/minitel_2_rpic.cpp:100]: Do you have a good reason for using macros?  In general, `enum` or `static constexpr` is preferred for constants, and `constexpr` functions are preferred for utility functions.  Macros provide worse compiler errors when there are issues, they produce less useful debug symbols, and function-like macros are error-prone due to the way substitution is done at preprocessing time (e.g. your macro here will behave unintuitively if the `bit` argument is an expression due to operator precedence rules).

File: src/mame/drivers/minitel_2_rpic.cpp [hunk 2]
```diff
@@ -92,6 +108,8 @@ class minitel_state : public driver_device
 		, m_ts9347(*this, "ts9347")
 		, m_palette(*this, "palette")
 		, m_i2cmem(*this, "i2cmem")
+		, m_modem(*this, "modport0")
+		, m_serport(*this, "serport0")
```
Comment (cuavas) [inline src/mame/drivers/minitel_2_rpic.cpp:112]: Slot device tags are user-facing.  How are the ports actually labelled on the machine?

File: src/mame/drivers/minitel_2_rpic.cpp [hunk 3]
```diff
@@ -200,25 +230,66 @@ void minitel_state::port1_w(uint8_t data)
...
+	{
+		tonedetect = 1;
+	}
+	else
+	{
+		tonedetect = 0;
+	}
+
+	// Main transmission mode :  PORT_1_MDM_RTS = 0, CTRL_REG_DTMF = 1, CTRL_REG_MCBC = 0
+	if( BIT(last_ctrl_reg, CTRL_REG_LINERELAY_BIT) )
```
Comment (cuavas) [inline src/mame/drivers/minitel_2_rpic.cpp:257]: Please don’t be inconsistent with spaces after commas like this.  We generally use space after comma.

File: src/mame/drivers/minitel_2_rpic.cpp [hunk 4]
```diff
@@ -430,6 +501,14 @@ void minitel_state::minitel2(machine_config &config)
 
 	TIMER(config, "minitel_sl", 0).configure_scanline(FUNC(minitel_state::minitel_scanline), "screen", 0, 10);
 
+	RS232_PORT(config, m_modem, default_rs232_devices, nullptr);
+	m_modem->rxd_handler().set_inputline("maincpu", MCS51_INT1_LINE).invert();
+
+	RS232_PORT(config, m_serport, default_rs232_devices, nullptr);
+	m_serport->rxd_handler().set_inputline("maincpu", MCS51_RX_LINE);
```
Comment (cuavas) [inline src/mame/drivers/minitel_2_rpic.cpp:508]: You have an object finder for the main CPU.

File: src/mame/drivers/minitel_2_rpic.cpp [hunk 5]
```diff
@@ -438,8 +517,29 @@ void minitel_state::minitel2(machine_config &config)
...
+	DEVICE_INPUT_DEFAULTS( "RS232_STOPBITS", 0xff, RS232_STOPBITS_1 )
+DEVICE_INPUT_DEFAULTS_END
+
+static DEVICE_INPUT_DEFAULTS_START( m_serport )
+	DEVICE_INPUT_DEFAULTS( "RS232_TXBAUD", 0xff, RS232_BAUD_9600 )
+	DEVICE_INPUT_DEFAULTS( "RS232_RXBAUD", 0xff, RS232_BAUD_9600 )
+	DEVICE_INPUT_DEFAULTS( "RS232_DATABITS", 0xff, RS232_DATABITS_8 )
+	DEVICE_INPUT_DEFAULTS( "RS232_PARITY", 0xff, RS232_PARITY_NONE )
+	DEVICE_INPUT_DEFAULTS( "RS232_STOPBITS", 0xff, RS232_STOPBITS_1 )
+DEVICE_INPUT_DEFAULTS_END
```
Comment (cuavas) [inline src/mame/drivers/minitel_2_rpic.cpp:541]: You aren’t attaching these to any slot options.  In fact, this shouldn’t actually compile due to unused file statics.

File: src/mame/drivers/rf51w14cg.cpp [hunk 6]
```diff
@@ -1,138 +1,138 @@
...
+	ROM_LOAD( "recreativos_franco_sonido-2_7c8bced_01083d.u4", 0x80000, 0x80000, CRC(cf8e7957) SHA1(5d30d7f15c1690b819e467fc308f12f97577b906) )
+
+	ROM_REGION( 0x2000, "nvram", 0 )
+	ROM_LOAD( "recreativos_franco_mod_rf51w14-cg_2003_m48t08.bin", 0x0000, 0x2000, NO_DUMP )
+ROM_END
+
+} // anonymous namespace
+
+//    YEAR  NAME       PARENT  MACHINE    INPUT      CLASS            INIT        ROT   COMPANY               FULLNAME                                                      FLAGS
+GAME( 2003, rf51w14cg, 0,      rf51w14cg, rf51w14cg, rf51w14cg_state, empty_init, ROT0, "Recreativos Franco", "unknown Recreativos Franco slot machine (model RF51W14-CG)", MACHINE_IS_SKELETON_MECHANICAL )
```
Comment (cuavas) [inline src/mame/drivers/rf51w14cg.cpp:138]: What have you done with this file?

**Review by cuavas — CHANGES_REQUESTED** [review]

File: src/mame/drivers/minitel_2_rpic.cpp (hunk 1 above)
Comment (jfdelnero) [reply src/mame/drivers/minitel_2_rpic.cpp]: fixed. both defines and macro replaced.

File: src/mame/drivers/minitel_2_rpic.cpp (hunk 2 above)
Comment (jfdelnero) [reply src/mame/drivers/minitel_2_rpic.cpp]: "modport0" changed to "modem" and "serport0" to "periinfo" 
Please note that the real name for "periinfo" is  "prise péri-informatique".

File: src/mame/drivers/minitel_2_rpic.cpp (hunk 3 above)
Comment (jfdelnero) [reply src/mame/drivers/minitel_2_rpic.cpp]: fixed

File: src/mame/drivers/rf51w14cg.cpp (hunk 6 above)
Comment (jfdelnero) [reply src/mame/drivers/rf51w14cg.cpp:138]: Nothing. I don't know why this file appears in this pull-request. Carriage return issue ?

File: src/mame/drivers/minitel_2_rpic.cpp (hunk 4 above)
Comment (jfdelnero) [reply src/mame/drivers/minitel_2_rpic.cpp]: fixed.

File: src/mame/drivers/minitel_2_rpic.cpp (hunk 5 above)
Comment (jfdelnero) [reply src/mame/drivers/minitel_2_rpic.cpp]: i "may" have fixed this issue in the last commit. Let me know if this is ok.

**Comment by jfdelnero** [comment]
Hello,

Most of the modifications in this pull-request was finally pushed through the @ovalenti pull request :

https://github.com/mamedev/mame/pull/9418

I close this one.

**Pull request closed without merging**